src/fsharp/vs/ServiceUntypedParse.fs

```diff
@@ -79,6 +79,8 @@ type CompletionContext =
     | ParameterList of pos * HashSet<string>
     | AttributeApplication
     | OpenDeclaration
+    /// completing pattern type (e.g. foo (x: |))
+    | PatternType
```

Comment:
Should this be named `ParameterType`?

Reply:
Looks like `TypeAnnotation` would be the better candidate of my two suggestions since it is also hit here:

```fsharp
let foo (s: string): ^
```

Reply:
No, this is for `xx: type` at any place, not just for parameters. And this thing is called pattern I think.

Reply:
Hmm, I think what you're referring to is the _[patterns that have type annotations](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/pattern-matching#parenthesized-pattern)_, so `PatternWithTypeAnnotation` would be the name I'd think of here.  That may have been what you meant, but when I read `PatternType` I thought it was referring to acompletion context which constituted a pattern of any kind.

Reply:
I believe Pattern refers to the bound pattern (when deconstructing a DU, a tuple, a record, etc.) and Type to the type annotation.

```fsharp
let foo ((Some s): string option) = ...
```

parameter can be a simple symbol but also can be a pattern.